src/server/scriptInfo.ts

```diff
@@ -574,7 +574,12 @@ namespace ts.server {
 
         markContainingProjectsAsDirty() {
             for (const p of this.containingProjects) {
-                p.markAsDirty();
+                if (this.cacheSourceFile) {

```

Comment:
Why does this depend on `this.cacheSourceFile` it could be that project doesn't use the same source file. Don't you want to do it all the time. Also do not rely on filename instead use `Path`

Reply:
Yeah, it should be all the time. I wasn’t sure the correct way to get the source file from here, but I’ll try getting it from the project somehow.

Reply:
This is fixed.